Sources/SolToBoogie/ProcedureTranslator.cs

```diff
@@ -2469,21 +2486,17 @@ private BoogieCallCmd MkFallbackDispatchCallCmd(FunctionCall node, BoogieExpr am
 
         private void TranslateSendCallStmt(FunctionCall node, BoogieIdentifierExpr returnExpr, BoogieExpr amountExpr)
         {
-            var guard = new BoogieBinaryOperation(BoogieBinaryOperation.Opcode.GE,
-                new BoogieMapSelect(new BoogieIdentifierExpr("Balance"), new BoogieIdentifierExpr("this")),
-                amountExpr);
-
-            // call FallbackDispatch(from, to, amount)
-            var callStmt = MkFallbackDispatchCallCmd(node, amountExpr);
-
-            var thenBody = new BoogieStmtList();
-            thenBody.AddStatement(callStmt); 
-            thenBody.AddStatement(new BoogieAssignCmd(returnExpr, new BoogieLiteralExpr(true)));
-
-            var elseBody = new BoogieAssignCmd(returnExpr, new BoogieLiteralExpr(false));
-

```

Comment:
Why are you removing this code region? This changes the default behavior even when your revert logic is not triggered.